src/maliput_malidrive/base/road_geometry.cc

```diff
@@ -733,6 +750,16 @@ const Lane* RoadGeometry::ApplyOffsetToLane(const Lane* initial_lane, int lane_o
   return target_lane;
 }
 
+maliput::math::RollPitchYaw RoadGeometry::OpenScenarioRoadPositionReferenceLineOrientation(
+    const OpenScenarioRoadPosition& xodr_road_position) const {
+  MALIDRIVE_THROW_UNLESS(xodr_road_position.road_id >= 0);
+  MALIDRIVE_THROW_UNLESS(xodr_road_position.s >= 0.);
+  const Segment* target_segment{this->FindSegmentByOpenScenarioRoadPosition(xodr_road_position)};
+  MALIPUT_THROW_UNLESS(target_segment != nullptr);
+  const double p = target_segment->road_curve()->PFromP(xodr_road_position.s);
+  return target_segment->road_curve()->Orientation(p);
+}
+
```

Comment:
It is important to mention that the road's reference line in the OpenDrive is a 2d line in the ground. In this case we are returning a full 3d orientation because at the road curve level we also care about the elevation and super elevation.
If this is ok then let's go ahead.

Reply:
I'm aware we are exposing a bit more than necessary here. I feel that our use case may take advantage from this, in a positive way I mean.